status-im/status-react
# feat: approve token transactions on swap

fixes #20343 

### Summary

This PR implements Approve transctions for ERC20 tokens when getting a Swap Proposal that needs the pay token to be approved before proceeding.

In the video below, an example of Approving 8 DAI on Mainnet

https://github.com/user-attachments/assets/0ea03a1e-a60c-4dd4-a2a6-8141a4d371ca

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account
- user profile updates
- networks
- mailservers
- fleet
- bootnodes

##### Non-functional

- battery performance
- CPU performance / speed of the app
- network consumption

### Steps to test

- Open Status
- Login
- Go to wallet
- Select an account
- Select Swap option
- Select a ERC20 token on Select asset to pay screen
- Enter a valid amount which is higher than the current approved amount for Paraswap contract
- Wait for swap proposal to appear
- At this stage, a valid amount appears on the receive token side, but "Review swap" button should be disabled because we need to first Approve / Set spending cap for the paying amount for the pay token.
- Tap on Approve
- Slide to execute the transction
- Enter password
- Wait for the approve transction to be confirmed and the status is shown correctly below the token input
- When the Approve transaction is confirmed, a toast should appear

status: ready
